lib/fizzy/leb128.hpp

```diff
@@ -41,7 +41,8 @@ std::pair<T, const uint8_t*> leb128s_decode(const uint8_t* input)
         if ((*input & 0x80) == 0)
         {
             // sign extend
-            if ((*input & 0x40) != 0)
+            if ((*input & 0x40) != 0 &&
+                static_cast<unsigned>(result_shift + 7) < sizeof(T_unsigned) * 8)
```

Comment:
Here you are comparing `unsigned` with `size_t` types. I think it would be better to change type of `result_shift` to `unsigned` or `size_t`.

Reply:
Yeah that's what I started with in the very first version :) will change